tests/suites/1_stateful/02_query/02_0004_attach_table.sh

```diff
@@ -29,4 +29,41 @@ echo "select * from table_to order by a;" | $MYSQL_CLIENT_CONNECT
 
 echo "delete from table_to where a=1;" | $MYSQL_CLIENT_CONNECT
 
+echo "rows after deletion"
 echo "select * from table_to order by a;" | $MYSQL_CLIENT_CONNECT
+
+
+# READ_ONLY attach
+
+echo "Attach READ_ONLY"
+# fetch table storage prefix
+storage_prefix=$(mysql -uroot -h127.0.0.1 -P3307  -e "set global hide_options_in_show_create_table=0;show create table table_from" | grep -i snapshot_location | awk -F'SNAPSHOT_LOCATION='"'"'|_ss' '{print $2}')
+
+# 1. READ_ONLY attach table
+echo "attach table table_read_only 's3://testbucket/admin/data/$storage_prefix' connection=(aws_key_id='minioadmin' aws_secret_key='minioadmin' endpoint_url='${STORAGE_S3_ENDPOINT_URL}') READ_ONLY;" | $MYSQL_CLIENT_CONNECT
+
+echo "check content of attach table"
+# ## check table content
+echo "select * from table_read_only order by a;" | $MYSQL_CLIENT_CONNECT
+
+# 2. READ_ONLY attach table should reflects the mutation of table being attached
+# del from the attachED table
+echo "delete from table_to where a=2;" | $MYSQL_CLIENT_CONNECT
+
+echo "check content of attach table, after row has been deleted from attachED table:"
+echo "  there should be only one row"
+# ## check table content
+echo "select * from table_read_only order by a;" | $MYSQL_CLIENT_CONNECT
+
+# 3. READ_ONLY attach table should aware of the schema evolution of table being attached
```

Comment:
We need `optimize table ... recluster/compact` WRITE related operations cases.

Reply:
yeah, working on it.

before that, a little refactoring is needed, FuseTable::check_mutable seems should be a method of Table, so that we can intercept the illegal mutations early and in a cleaner way,